### DISCUSSION TITLE
Doubt regarding the execution of scripts

### DISCUSSION THREAD

**Discussion by Ananthumpillai** [GrapesJS/grapesjs#5833] [General]
On preview mode I am adding a header to my page, this is actually working, but the variable headerHtml is having some scripts, which controls the hamburger menu and all, but those scripts are not getting executed.

domComponents.addComponent(
        {
          tagName: "div",
          content: `${headerHtml}<style>${headerCss}</style>`,
        },
        { at: 0 }
      );
      
      
      Is there any way to include those scripts in the component

**Comment by artf** [comment]
Please follow the documentation on how to handle properly [components with scripts](https://grapesjs.com/docs/modules/Components-js.html).

**Reply by Ananthumpillai** [reply]
Hi @artf,
I have gone through the page but it didn't solved my issue.
what I am actually trying is to add a header to the page on preview,

domComponents.addComponent(
{
tagName: "div",
content: ${headerHtml}<style>${headerCss}</style>,
},
{ at: 0 }
);

here headerHtml is having the body and scripts for the header, currently the scripts inside this is not executing. 


  document.body.appendChild(script());
   window.eval(script())
   I have extracted the script from the html and tried these approaches but it didn't worked.
   
It would be great if you could solve this issue.

Thanks
